### PR TITLE
Display progress when server is starting #390

### PR DESCRIPTION
**Pull request by piotrpekala7**

### CODE REVIEW

**Comment by ziajka** [comment]
Remote servers should be accessible even without verification they are running: 

Screenshot_2019-06-04_11-55-30

File: src/app/components/servers/servers.component.ts [hunk 1]
```diff
@@ -98,10 +98,12 @@ export class ServersComponent implements OnInit, OnDestroy {
   }
 
   async startServer(server: Server) {
+    server.status = "starting";
```
Comment (ziajka) [inline src/app/components/servers/servers.component.ts]: Shouldn't we rely on serverManagement service?

File: src/app/components/servers/servers.component.ts [hunk 2]
```diff
@@ -98,10 +98,12 @@ export class ServersComponent implements OnInit, OnDestroy {
...
     await this.serverManagement.start(server);
   }
 
   async stopServer(server: Server) {
+    server.status = "stopped";
```
Comment (ziajka) [inline src/app/components/servers/servers.component.ts]: Shouldn't we rely on serverManagement service?

File: src/app/validators/RotationValidator.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class RotationValidator {
+  get(control) {
+    if (+control.value>-360 && +control.value<=360) {
```
Comment (ziajka) [inline src/app/validators/RotationValidator.ts]: Math.abs

Comment (piotrpekala7) [reply src/app/validators/RotationValidator.ts]: Cannot be applied, value in (-360, 360>

File: src/app/components/servers/servers.component.ts (hunk 2 above)
Comment (piotrpekala7) [reply src/app/components/servers/servers.component.ts]: Removed

File: src/app/components/servers/servers.component.ts (hunk 1 above)
Comment (piotrpekala7) [reply src/app/components/servers/servers.component.ts]: Well, server is not yet running but we sent request. I dunno what you mean by that, we rely on serverManagementService

Comment (ziajka) [reply src/app/components/servers/servers.component.ts]: await this.serverManagement.start(server); should update.status to "starting" and propagate changes to UI.

Comment (piotrpekala7) [reply src/app/components/servers/servers.component.ts]: The status will be then changed to 'started'

Comment (piotrpekala7) [reply src/app/components/servers/servers.component.ts]: As I understand introducing 'starting' status is not desirable

Comment (ziajka) [reply src/app/components/servers/servers.component.ts]: It is, but maybe in serverManagement service, not in ServersComponent

Comment (piotrpekala7) [reply src/app/components/servers/servers.component.ts]: Done

**Pull request merged**